spec/solver_spec.rb

```diff
@@ -0,0 +1,20 @@
+require './solver'
+
+describe 'Solver' do
+  context 'Testing class Solver' do
+    it 'Should return factorial of a number, reverse of a string and  print fizz, buzz and fizzbuzz' do
+      solver_mock = double('solver')
+      solver_class = Solver.new
+      allow(solver_mock).to receive(:factorial).with(3).and_return(6)
+      allow(solver_mock).to receive(:reverse).with('hello').and_return('olleh')
+      allow(solver_mock).to receive(:fizzbuzz).with(3).and_return('fizz')
+      allow(solver_mock).to receive(:fizzbuzz).with(5).and_return('buzz')
+      allow(solver_mock).to receive(:fizzbuzz).with(15).and_return('fizzbuzz')
+      expect(solver_class.factorial(3)).to eq(6)
```

Comment:
- [ ] For the factorial method, you only covered one edge case which is a positive number.
Please make your test cover all edge cases like this method suppose to raise an exception as an expectation when given a negative number and the factorial of `0` is equal to `1`.